code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -219,6 +220,10 @@
 		return
 
 	module.transform_to(modulelist[input_module])
+	if(module)
+		for(var/trait in module.traits)
+			var/mob/living/silicon/robot/R = loc
+			ADD_TRAIT(R, trait, "Cyborg_Module")
```

Comment:
Do you remove the trait when the module is removed?

Reply:
I do now.